micro-rdk/src/common/ota.rs

```diff
@@ -256,31 +269,22 @@ impl<S: OtaMetadataStorage> OtaService<S> {
             pending_version,
             max_size,
             address,
+            needs_reboot: false,
```

Comment:
I think it's better to have `update` return a boolean as to whether an update was actually performed rather than store state?